### ISSUE TITLE
Engine: move F* backend to the generic printer

### ISSUE THREAD

**Issue by W95Psp** [hacspec/hax#942]
The current printer is huge and hard to maintain.
It also uses the internal AST from F*, which is nice because it's giving us a pretty printer for free. But in the same times, that makes things complicated, the pretty printer isn't all that pretty, and we cannot transport spans.

**Comment by franziskuskiefer** [comment]
Looks like nothing happened here this week. @W95Psp do you want to push it into next week or what's the plan?

**Comment by W95Psp** [comment]
Yes, this week was awful (because of the migraine), so I did almost nothing :(
I probably won't spend time on that today, let's push to next week.
